core/domain/suggestion_services.py

```diff
@@ -2904,3 +2908,333 @@ def enqueue_contributor_ranking_notification_email_task(
+        template = _generate_translation_contributor_certificate(
+            language_code, from_date, to_date, username, user_id, date)
+
+    elif suggestion_type == feconf.SUGGESTION_TYPE_ADD_QUESTION:
+        template = _generate_question_contributor_certificate(
+            from_date, to_date, username, user_id, date)
+
+    else:
+        raise Exception('The suggestion type is invalid.')
+        
+    return _generate_contributor_certificate_image(template)
+
+
+def _generate_translation_contributor_certificate(
+    language_code: str,
+    from_date: datetime.datetime,
+    to_date: datetime.datetime,
+    username: str,
+    user_id: str,
+    date: str
+) -> str:
+    """Generates a certificate for a user's translation contributions.
+
+    Args:
+        language_code: str|None. The language for which the contributions should
+            be considered.
+        suggestion_type: str. The type of suggestion that the certificate
+            needs to generate.
+        from_date: datetime.datetime. The start of the date range for which
+            the contributions were created.
+        to_date: datetime.datetime. The end of the date range for which
+            the contributions were created.
+        username: str. The username of the contributor.
+        user_id: str. The user ID of the contributor.
+        date: datetime.datetime. The current date that should be included in the
+            certificate.
+
+    Returns:
+        str. The HTML template to generate the certificate.
+
+    Raises:
+        Exception. The language is invalid.
+    """
+    signature = feconf.TRANSLATION_TEAM_LEAD
+
+    language = next(filter(
+        lambda lang: lang['id'] == language_code,
+        constants.SUPPORTED_AUDIO_LANGUAGES), None)
+    if not language:
+        raise Exception('The provided language is invalid.')
+    language_description = language['description']
+
+    suggestions = (
+        suggestion_models.GeneralSuggestionModel
+            .get_translation_suggestions_submitted_within_given_dates(
+                from_date, to_date, user_id, language_code))
+
+    words_count = 0
+    for model in suggestions:
+        suggestion = get_suggestion_from_model(model)
+
+        # Retrieve the html content that is emphasized on the 
+        # Contributor Dashboard pages. This content is what stands
+        # out for each suggestion when a user views a list of
+        # suggestions.
+        get_html_representing_suggestion = (
+            SUGGESTION_EMPHASIZED_TEXT_GETTER_FUNCTIONS[
+                suggestion.suggestion_type]
+        )
+        plain_text = _get_plain_text_from_html_content_string(
+            get_html_representing_suggestion(suggestion))
+
+        words = plain_text.split(' ')
+        words_without_empty_strings = [
+            word for word in words if word != '']
+        words_count += len(words_without_empty_strings)
+    # Source: https://github.com/oppia/oppia/pull/16513/files#r1027109026
```

Comment:
Can you point to the sheet directly?

Reply:
Ok

Reply:
Can you use a different link? I don't see a comment when using the one that you used. Also you can link the sheet directly, we allow more than 80 chars for links.